src/topology/algebra/ordered.lean

```diff
@@ -1831,3 +1831,31 @@ begin
       have : ∀ b, f b < a' ↔ f b - a < ε, by { intro b, simp [lt_sub_iff_add_lt] },
       simpa only [this] }}
 end
+
+lemma tendsto_at_top_supr_of_bdd {ι  : Type*} [nonempty ι] [semilattice_sup ι]
```

Comment:
This lemma should be named `tendsto_at_top_csupr`, and `tendsto_at_top_supr` should be a version for `complete_linear_order` not assuming `bdd_above (range f)`. BTW, do you need `linear_order` here?

Reply:
Okay I'll change the names. About `linear_order`, I have to admit I don't fully understand all the vocabulary here (first year uni) so maybe we should ask to @PatrickMassot as he did this proof

Reply:
I didn't think carefully about the minimal requirement here. `exists_lt_of_lt_csupr` is the reason I'm having this type class, but I'm super interested in thinking about circumventing this. We should probably have some list of examples and counter-examples of all those type classes (including the zoo from topology/algebra/ordered).

Reply:
I pushed a proof for two `preorder`s.

Reply:
This lemma needs a `linear_order`, I'll push a "fix compile" commit with a counter-example in a comment.